core/benchmark/storage/src/main.rs

```diff
@@ -1638,23 +1638,27 @@ impl TxReplayer {
         });
 
         TxReplayer {
-            storage_manager: storage_manager,
+            storage_manager,
             tx_counts: Cell::new(0),
             ops_counts: Cell::new(0),
-            exit: exit,
+            exit,
         }
     }
 
-    pub fn commit(mut latest_state: StateDb, txs: u64, ops: u64) -> H256 {
+    pub fn commit(
+        latest_state: &mut StateDb, last_state_root: &mut H256, txs: u64,
+        ops: u64,
+    )
+    {
         warn!("Committing block at tx {}, ops {}.", txs, ops);
-        let state_root =
-            latest_state.get_storage_mut().compute_state_root().unwrap();
-        latest_state
-            .get_storage_mut()
-            .commit(state_root.clone())
-            .unwrap();
 
-        state_root
+        // We want to use the delta trie root, but we don't want to compute
+        // twice.
+        let storage =
+            unsafe { std::mem::transmute::<_, &mut Storage>(latest_state) };
+        let state_root = storage.compute_state_root().unwrap();
+        storage.commit(state_root.state_root.delta_root).unwrap();
+        *last_state_root = state_root.state_root.delta_root;
```

Comment:
I think you can just use the code on the left and replace state_root with state_root.state_root.delta_root at 2 places.
And is it not necessary to pass last_state_root by mut reference?